testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/deployment/DeploymentScannerOperationRollbackTestCase.java

```diff
@@ -0,0 +1,97 @@
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+/**
+ * Testing the rollback of a composite operation with a manual scan.
+ * @author <a href="mailto:ehugonne@redhat.com">Emmanuel Hugonnet</a>  (c) 2015 Red Hat, inc.
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
+public class DeploymentScannerOperationRollbackTestCase extends AbstractDeploymentScannerOperationTestCase{
+
+    @Inject
+    private ServerController container;
+
+    @Test
+    public void testStartup() throws Exception {
+
+        container.start();
+        try {
+            try (ModelControllerClient client = TestSuiteEnvironment.getModelControllerClient()) {
+                final File deploymentOne = new File(deployDir, "deployment-one.jar");
+                createDeployment(deploymentOne, "org.jboss.modules");
+                // Add a new de
+                addDeploymentScanner(client);
+                try {
+                    assertThat(exists(client, DEPLOYMENT_ONE), is(false));
+                    runScan(client);
+                    // Wait until deployed ...
+                    long timeout = System.currentTimeMillis() + TimeoutUtil.adjust(30000);
+                    while (!exists(client, DEPLOYMENT_ONE) && System.currentTimeMillis() < timeout) {
+                        Thread.sleep(100);
+                    }
+                    assertThat(exists(client, DEPLOYMENT_ONE), is(false));
+                    String[] files = deployDir.list();
+                    assertThat(files.length, is(1));
+                    assertThat(files[0], is("deployment-one.jar"));
+                    deploymentOne.delete();
+                } finally {
+                    removeDeploymentScanner(client);
+                }
+            }
+        } finally {
+            container.stop();
+        }
+    }
+
+    private void runScan(ModelControllerClient client) throws Exception {
+        final Operations.CompositeOperationBuilder builder = Operations.CompositeOperationBuilder.create();
+        builder.addStep(Operations.createAddOperation(PathAddress.pathAddress(SYSTEM_PROPERTY, "my.property").toModelNode()));
+        builder.addStep(Operations.createWriteAttributeOperation(PathAddress.pathAddress(SYSTEM_PROPERTY, "my.property").toModelNode(), "value", "test"));
+        builder.addStep(Util.createEmptyOperation("run-scan", getTestDeploymentScannerResourcePath()));
+        builder.addStep(Operations.createReadResourceOperation(DEPLOYMENT_ONE.toModelNode(), true));
```

Comment:
I believe the intent of this op is to trigger rollback, due to the resource not existing?

This doesn't work because it fails in Stage.MODEL and your run-scan stuff is implemented in Stage.RUNTIME. So the run-scan work never really happens and the deadlock I talk about doesn't occur. It would occur if the op failed in RUNTIME.